strawberryfields/ops.py

```diff
@@ -594,9 +594,10 @@ def _apply(self, reg, backend, **kwargs):
         r = par_evaluate(self.p[0])
         phi = par_evaluate(self.p[1])
 
-        tf_complex = any(hasattr(arg, "numpy") and np.iscomplex(arg.numpy()) for arg in [r, phi])
+        np_args = [arg.numpy() if hasattr(arg, "numpy") else arg for arg in [r, phi]]
+        is_complex = any([np.iscomplexobj(np.real_if_close(arg)) for arg in np_args])
```

Comment:
Note that we need the `np.real_if_close()`, otherwise the `integration/test_algorithms.py::test_gaussian_gate_teleportation` test will fail. The test fails on the line:
```python
Xgate(s * q[0].par) | q[2]
```
Here, `q[0].par` is a real-valued object with a complex dtype.

Reply:
Thanks for catching this Tom; so this only failed on Python 3.6?